### ISSUE TITLE
Improve error handling when Personal Access Token is not specified in configuration

### ISSUE THREAD

**Issue by jeremy-jameson** [RickStrahl/GistIntegration-MarkdownMonster-Addin#5]
I might have stumbled upon the same issue as **gep13** (#3 - **No Gist being created**) or I might have encountered something different due to my environment.

After setting **GithubUserToken** and **GithubUsername** in my **PasteCodeAsGistAddin.json** file, I attempted to save a new Gist. It appeared that some error occurred since the following warning was displayed at the bottom of the **Save Document to Gist** window:

:warning: *Gist was not saved - failed to save as Gist.*

I ended up downloading the source for the add-in and debugging the issue. This is where things get a little confusing.

It appears that when you uninstall the add-in (using **Tools** :arrow_right: **Add-in Manager**), the config file is reset to the following:

```Json
{
  "GithubUserToken": null,
  "GithubUsername": null
}
```

Consequently, when I subsequently ran through the repro steps in the debugger (using my own build of the add-in), I experienced the same warning message shown above (i.e. the Gist was not saved) but probably for a different reason -- since this time around, there was definitely no token specified in the config file.

I noticed the add-in catches any exception that occurs while posting the Gist and specifies the exception message on the `gist` object:

```csharp
try
{
    result = HttpUtils.HttpRequestString(settings);
}
catch (Exception ex)
{
    gist.hasError = true;
    gist.errorMessage = "Gist upload failed: " + ex.Message;
    return gist;
}
```

However, it does not appear this error message is displayed anywhere.

When I originally encountered this issue, I kept looking for some kind of error in the log (via **Help** :arrow_right: **Show Error Log**), but it kept saying there were no errors.

I recommend writing the exception message above in the log and also adding the following near the top of **GistClient.PostGist**:

```csharp
if (string.IsNullOrEmpty(githubUserToken))
{
    gist.hasError = true;
    gist.errorMessage = "GitHub user token is not specified.";
    return gist;
}
```

More specifically, the above code should be added after the following:

```csharp
if (string.IsNullOrEmpty(githubUserToken))
    githubUserToken = PasteCodeAsGistConfiguration.Current.GithubUserToken;
```

Also note that the current behavior of automatically resetting the status message at the bottom of the **Save Document to Gist** window (after a few seconds) is potentially confusing as well. I would expect any error or warning message to stay visible until I have either acknowledged it (for example, by canceling/closing the window) or attempted the operation again (by clicking the save button -- perhaps thinking this is some sort of transient issue).

**Comment by jeremy-jameson** [comment]
I might have figured out the crux of the original issue (i.e. missing GitHub token). If my hunch is correct, it is due to the use of Local vs. Roaming AppData (i.e. roaming profiles). My Active Directory account is configured to use a roaming profile (stored on a file server and accessed through Redirected Folders/Offline Files). I've seen a couple of places in the code where different AppData locations are used. It might just be in the assembly references (which would only affect development), but it could be something more problematic. Let me know if you want me to dig into it.

**Comment by RickStrahl** [comment]
Thank you for your work on this - this is very helpful. It's actually on my todo list for this addin but that list has kind of fallen by the way side, so thank you. 

Addressed with merged PR #7 .
